GeeksCoreLibrary/Components/Configurator/Services/ConfiguratorsService.cs

```diff
@@ -763,6 +773,398 @@ private List<int> GetStepPosition(DataRow[] allSteps, DataRow stepData, List<int
+                stepData.Options = options;
+                return;
+            }
+            
+            // Get the question from the response.
+            JToken responseData = restResponse.Content.StartsWith("{") ? JObject.Parse(restResponse.Content) : JArray.Parse(restResponse.Content);
+            var keyParts = new List<string>(questionKey.Split('.'));
+            JToken currentObject = responseData;
+
+            // Step into the object till the correct object is found or the path leads to an invalid object.
+            while (keyParts.Count > 0)
+            {
+                if (keyParts[0].StartsWith("["))
+                {
+                    var selector = keyParts[0].Substring(1, keyParts[0].Length - 2);
+                    var selectorParts = selector.Split('=');
+
+                    var matchFound = false;
+
+                    foreach (JToken jToken in currentObject)
+                    {
+                        if (jToken.Type != JTokenType.Object || !jToken[selectorParts[0]].Value<string>().Equals(selectorParts[1], StringComparison.InvariantCultureIgnoreCase))
+                        {
+                            continue;
+                        }
+
+                        currentObject = jToken;
+                        matchFound = true;
+                        break;
+                    }
+                    
+                    // If no match is found set the current object to null so the question is not found and will not have answers.
+                    if (!matchFound)
+                    {
+                        currentObject = null;
+                    }
+                }
+                else
+                {
+                    currentObject = currentObject[keyParts[0]];
+                }
+                
+                keyParts.RemoveAt(0);
+                
+                // If the current object is null, the question is not found and will not have answers.
+                if (currentObject == null)
+                {
+                    stepData.Options = options;
+                    return;
+                }
+            }
+
+            if (!String.IsNullOrWhiteSpace(stepData.MinimumValue) && stepData.MinimumValue.StartsWith("JSON:", StringComparison.InvariantCultureIgnoreCase))
+            {
+                stepData.MinimumValue = currentObject.SelectToken(stepData.MinimumValue.Substring(5)).Value<string>();
+                
+                // If the property could not be found while being expected the question is not complete and will not have answers.
+                if (stepData.MinimumValue == null)
+                {
+                    stepData.Options = options;
+                    return;
+                }
+            }
+            if (!String.IsNullOrWhiteSpace(stepData.MaximumValue) && stepData.MaximumValue.StartsWith("JSON:", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Please add a newline between 2 separate if statements, otherwise it looks too much like an else or elseif at first glance.